solver/src/board.rs

```diff
@@ -54,16 +56,40 @@ impl Board {
         Self { board }
     }
 
-    pub fn new_from_day_pos(month_pos: Point, day_pos: Point) -> Self {
-        let mut walls = [
-            Point::new(0, 6),
-            Point::new(1, 6),
-            Point::new(6, 3),
-            Point::new(6, 4),
-            Point::new(6, 5),
-            Point::new(6, 6),
-        ]
-        .iter()
+    pub fn new_from_day_pos(month_pos: Point, day_pos: Point, puzzle_type: PuzzleType ) -> Self {
+        let mut walls = match puzzle_type {
+			PuzzleType::DragonFjord => {
+			[
+				Point::new(0, 6),
+				Point::new(1, 6),
+				Point::new(6, 3),
+				Point::new(6, 4),
+				Point::new(6, 5),
+				Point::new(6, 6),
+			]
+			}
+			PuzzleType::JarringWords => {
+			[
+				Point::new(0, 6),
+				Point::new(1, 6),
+				Point::new(6, 3),
+				Point::new(6, 4),
+				Point::new(6, 5),
+				Point::new(6, 6),
+			]
+			}
+			PuzzleType::Tetromino => {
+			[
+				Point::new(0, 6),
+				Point::new(1, 6),
+				Point::new(6, 0),
+				Point::new(6, 1),
+				Point::new(6, 2),
+				Point::new(6, 3),
+			]
+			}
+
+		}.iter()
```

Comment:
Since `DragonFjord` and `JarringWords` use the same board, can you merge these arms with `|` operator?
```
match puzzle_type {
    PuzzleType::DragonFjord | PuzzleType::JarringWords => ...,
   PuzzleType::Tetromino => ....
}
```

Reply:
Done!